src/app.py

```diff
@@ -829,6 +864,22 @@ def publish_datastage(identifier):
                     return jsonify({"error": f"{dataset_uuid} missing contacts or contributors. Must have at least one of each"}), 400
             ingest_helper = IngestFileHelper(app.config)
 
+            # Save a .json file with the metadata information at the top level directory...
+            if dataset_ingest_matadata_dict is not None:
+                json_object = json.dumps(dataset_ingest_matadata_dict['metadata'], indent=4)
+                json_object += '\n'
+                ds_path = ingest_helper.dataset_directory_absolute_path(dataset_data_access_level,
+                                                                        dataset_group_uuid, uuid, False)
+                md_file = os.path.join(ds_path, "metadata.json")
+                logger.info(f"publish_datastage; writing md_file: '{md_file}'; "
+                            f"containing ingest_matadata.metadata: '{json_object}'")
+                try:
+                    with open(md_file, "w") as outfile:
+                        outfile.write(json_object)
+                except IOError as ioe:
+                    logger.exception(f"Error while writing md_file {md_file}; {ioe}")
+                    return jsonify({"error": f"{dataset_uuid} problem writing json file."}), 400
```

Comment:
Related to above comment, if it turns out be a `FileNotFoundError: [Errno 2] No such file or directory`, it means something wrong on the server side, not necessarily 400 client error.

Reply:
Right.... So, I'll change it to a 500.